domains/certificates/RegistryConnector/Worker.UnitTests/EventHandlers/MeasurementEventHandlerTests.cs

```diff
@@ -0,0 +1,116 @@
+    IClassFixture<RegistryConnectorApplicationFactory>
+{
+    private readonly RegistryConnectorApplicationFactory factory;
+    private readonly DbContextOptions<ApplicationDbContext> options;
+
+    public MeasurementEventHandlerTests(
+        PostgresContainer dbContainer,
+        RabbitMqContainer rabbitMqContainer,
+        RegistryConnectorApplicationFactory factory)
+    {
+        this.factory = factory;
+        factory.RabbitMqOptions = rabbitMqContainer.Options;
+        factory.ConnectionString = dbContainer.ConnectionString;
+        factory.Start();
+
+        options = new DbContextOptionsBuilder<ApplicationDbContext>().UseNpgsql(dbContainer.ConnectionString).Options;
+        using var dbContext = new ApplicationDbContext(options);
+        dbContext.Database.EnsureCreated();
+    }
+
+    [Fact]
+    public async Task ShouldOnlyProcessMessagesOnSaveChanges()
+    {
+        var privateKey = new Secp256k1Algorithm().GenerateNewPrivateKey();
+        var hdPublicKey = privateKey.Derive(1).Neuter();
+        var gsrn = GsrnHelper.GenerateRandom();
+        var subject = Guid.NewGuid().ToString();
+        var now = DateTimeOffset.UtcNow;
+        var utcMidnight = now.Subtract(now.TimeOfDay);
+
+        var measurement = new EnergyMeasuredIntegrationEvent(
+            GSRN: gsrn,
+            DateFrom: utcMidnight.ToUnixTimeSeconds(),
+            DateTo: utcMidnight.AddHours(1).ToUnixTimeSeconds(),
+            Quantity: 42,
+            Quality: MeasurementQuality.Measured);
+
+        using (var scope = factory.ServiceScope())
+        {
+            using (var dbContext = scope.ServiceProvider.GetRequiredService<ApplicationDbContext>())
+            {
+                dbContext.Contracts.Add(new CertificateIssuingContract
+                {
+                    MeteringPointType = MeteringPointType.Consumption,
+                    ContractNumber = 1,
+                    Created = DateTimeOffset.UtcNow,
+                    EndDate = null,
+                    StartDate = utcMidnight,
+                    GSRN = gsrn,
+                    GridArea = "DK1",
+                    Id = Guid.NewGuid(),
+                    MeteringPointOwner = subject,
+                    Technology = null,
+                    WalletUrl = "https://foo",
+                    WalletPublicKey = hdPublicKey.Export().ToArray(),
+                });
+
+                await scope.ServiceProvider.GetRequiredService<IPublishEndpoint>().Publish(measurement);
+
+                //We wait and confirm that the publish does not happen before SaveChanges has been called
+                //(i.e. before the message has been put in the database)
+                await Task.Delay(TimeSpan.FromSeconds(15));
+
+                await dbContext.SaveChangesAsync();
+            }
+        }
+
+        (await QueryUntilConsumptionCertificateFound(gsrn)).Should().Be(true);
+    }
+
+    private async Task<bool> QueryUntilConsumptionCertificateFound(string gsrn)
```

Comment:
Could this be renamed to something like `isConsumptionCertificateSaved`? I find it hard to read what the test does and why it does it.
Just reading `QueryUntilCunsomption....` i wouldnt expect it to return a bool

Reply:
Renamed